### PR TITLE
cmd, internal/*, node: fix resource leaks

### LINKED ISSUE #8892: node: resource leaks in some error conditions
**Tendermint version** (use `tendermint version` or `git rev-parse --verify HEAD` if installed from source):
5f5e74798bcc6045bb454a082405145a3230ab1a

**What happened**:
Noticed while doing security audits that there is a bunch of code in node that doesn't invoke .Close on resources like in:
* makeNode
* initDBs
* createRouter
* createAndStartPrivValidatorSocketClient
* createAndStartPrivValidatorGRPCClient

**What you expected to happen**:
All resources released.

**Have you tried the latest version**: yes

### LINKED ISSUE #8893: internal/*: resource leaks in some error conditions
**Tendermint version** (use `tendermint version` or `git rev-parse --verify HEAD` if installed from source):
5f5e74798bcc6045bb454a082405145a3230ab1a

**What happened**:
Noticed while doing security audits that there is a bunch of code in node that doesn't invoke .Close on resources like in:
* consensus.newConsensusStateForReplay
* internal.NewFromConfig
* libs.(*Group).NewReader
* p2p.(*MemoryNetwork).CreateTransport

**What you expected to happen**:
All resources released.

**Have you tried the latest version**: yes

### PR DESCRIPTION
**Pull request by odeke-em**
Fixes resources leaks found internally at Orijtech Inc during
security audits by our cyber team.
Fixing these used the technique of named error return values
that check if the return value was an non-nil error and if so
close the target resource.

Fixes #8892
Fixes #8893

### CODE REVIEW

File: cmd/tendermint/commands/reindex_event.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ func loadEventSinks(cfg *tmcfg.Config) ([]indexer.EventSink, error) {
 	return eventSinks, nil
 }
 
-func loadStateAndBlockStore(cfg *tmcfg.Config) (*store.BlockStore, state.Store, error) {
+func loadStateAndBlockStore(cfg *tmcfg.Config) (_ *store.BlockStore, _ state.Store, rerr error) {
```
Comment (tychoish) [inline cmd/tendermint/commands/reindex_event.go:144]: these `rerr` values are not set anywhere, which makes this change a noop. I think you need to declare this as `err` and then be careful to not re-declare it later.

**Review by tychoish — CHANGES_REQUESTED** [review]
While on  the surface these changes seem unobjectionable, I don't think that they can actually change anything (because they check error values that are never mutated), or protect against resource leaks that can't happen in practice (e.g. resource leaks before panics in test fixtures, or during node before the node crashes.) (or both!)

File: node/setup.go [hunk 2]
```diff
@@ -83,6 +83,12 @@ func initDBs(
 	closers := []closer{}
 	blockStore := store.NewBlockStore(blockStoreDB)
 	closers = append(closers, blockStoreDB.Close)
+	defer func() {
+		if rerr != nil {
+			blockStore.Close()
+			_ = closeFn()
+		}
+	}()
```
Comment (cmwaters) [inline node/setup.go:91]: Isn't the function calling this responsible for also closing the resources if there's an error (and not this function)

File: cmd/tendermint/commands/reindex_event.go (hunk 1 above)
Comment (tychoish) [reply cmd/tendermint/commands/reindex_event.go:144]: I know how the defer statement works, I'm saying there is no code (as in the example,) in these patches which modifies this line? Theoretically something could, but nothing does.

File: internal/p2p/transport_memory.go [hunk 3]
```diff
@@ -49,6 +49,7 @@ func (n *MemoryNetwork) CreateTransport(nodeID types.NodeID) *MemoryTransport {
 	n.mtx.Lock()
 	defer n.mtx.Unlock()
 	if _, ok := n.transports[nodeID]; ok {
+		t.Close()
```
[the remark this answers is not in the document]
Comment (tychoish) [reply internal/p2p/transport_memory.go:52]: Perhaps, but again, this is somewhat confusingly in test code, and given the panic it's pretty low impact.

**Pull request closed without merging**